core/eolearn/core/eodata.py

```diff
@@ -255,6 +255,23 @@ def add_feature(self, feature_type, feature_name, value):
         """
         self._check_if_dict(feature_type)
         self[feature_type][feature_name] = value
+    
+    def rename_feature(self,feature_type,feature_name,new_feature_name):
+        """Renames the feature ``feature_name`` to ``new_feature_name`` from dictionary of ``feature_type``.
+
+        :param feature_type: Enum of the attribute we're about to modify
+        :type feature_type: FeatureType
+        :param feature_name: Name of the feature of the attribute
+        :type feature_name: str
+        :param new_feature_name : New Name of the feature of the attribute
+        :type feature_name: str
+        """
+        LOGGER.debug("Renaming feature '%s' from attribute '%s'", feature_name, feature_type.value)

```

Comment:
I would put the debug message after the check is completed so that it really does that, and additionally would print and error message if the conditions are not satisfied